examples/no_std.rs

```diff
@@ -108,7 +108,7 @@ mod no_std {
         let method: HttpMethod = opts.method.parse()?;
         let url: Url = opts.url.parse()?;
 
-        let s = MyFakeTcpStream::connect(url.authority.as_ref(), url.port()?)?;
+        let s = MyFakeTcpStream::connect(url.host_str().unwrap(), url.port_or_known_default().unwrap())?;
```

Comment:
This `unwrap` here and the other places I'm not okay with. In these examples I want them to be more like you would have a real program, and panic on non-HTTP urls isn't good, we need to instead return an error like you did in the other place, I suspect that you didn't because it can get a bit long, I think we should find a more ergonomic way to check for this.

Reply:
done by add HttpUrl struct